### PR TITLE
Revert "Updated _evaluate_pipelines to consolidate side effects"

### PR DESCRIPTION
**Pull request by christopherbunn**
Reverts alteryx/evalml#1337

### CODE REVIEW

**Comment by dsherry** [comment]
Yep, due to the test flake from #1408

**Review by dsherry — APPROVED** [review]
👍

**Review by freddyaboulton — APPROVED** [review]
Thanks @christopherbunn  ! 👍

**Comment by dsherry** [comment]
@christopherbunn for the release notes, I'd say you can just append "reverted :pr:`1409`" to the end. Then when we fix and re-merge, we can delete that entry and replace with the new one. That work?

**Comment by christopherbunn** [comment]
> @christopherbunn for the release notes, I'd say you can just append "reverted :pr:`1409`" to the end. Then when we fix and re-merge, we can delete that entry and replace with the new one. That work?

@dsherry Since the changelog line for 1337 isn't there anymore with this revert PR, I think I'll just make a new entry for now. We can delete when we push it back in.

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Release Notes
     * Fixes
         * Updated enum classes to show possible enum values as attributes :pr:`1391`
     * Changes
+        * Reverted changes from :pr:`1337` :pr:`1409`
```
Comment (dsherry) [inline docs/source/release_notes.rst:11]: 👍  cool, yep we can delete this when we merge a new PR

**Pull request merged**